lib/archethic/transaction_chain.ex

```diff
@@ -994,4 +1036,24 @@ defmodule Archethic.TransactionChain do
         {:error, :network_issue}
     end
   end
+
+  @doc """
+  Retrieve the First transaction address for a chain from P2P Quorom
+  """
+  @spec fetch_first_transaction_address_remotely(address :: binary()) ::
+          {:ok, binary()} | {:error, :network_issue}
+  def fetch_first_transaction_address_remotely(address) when is_binary(address) do
+    nodes = Election.chain_storage_nodes(address, P2P.authorized_and_available_nodes())
+
+    case P2P.quorum_read(nodes, %GetFirstTransactionAddress{address: address}) do
+      {:ok, %FirstTransactionAddress{address: ^address}} ->
```

Comment:
This case should be `{:ok, %NotFound{}}` I guess as it is returned by the message when the first address is not retrieved